app/payment-link-v2/reference/get-index.controller.js

```diff
@@ -0,0 +1,37 @@
+'use strict'
+
+const { response } = require('../../utils/response')
+const { getSessionVariable } = require('../../utils/cookie')
+const { NotFoundError } = require('../../errors')
+const getBackLinkUrl = require('./get-back-link-url')
+
+module.exports = (req, res, next) => {
+  const product = req.product
+
+  const data = {
+    productExternalId: product.externalId,
+    productName: product.name
+  }
+
+  if (!product.reference_enabled) {
+    return next(new NotFoundError('Attempted to access reference page with a product that auto-generates references.'))
+  }
+
+  const sessionReferenceNumber = getSessionVariable(req, 'referenceNumber')
+
+  data.backLinkHref = getBackLinkUrl(sessionReferenceNumber, product)
+
+  if (sessionReferenceNumber) {
+    data.referenceNumber = sessionReferenceNumber
+  }
+
+  if (product.reference_label) {
+    data.paymentReferenceLabel = product.reference_label
+  }
+
+  if (product.reference_hint) {
```

Comment:
Probably don't need this `if`, just include the `reference_hint` in the `data` object always and if it's blank, the template will deal with that.

Reply:
Good point - done